components/browser/state/src/main/java/mozilla/components/browser/state/state/BrowserState.kt

```diff
@@ -39,5 +41,6 @@ data class BrowserState(
     val downloads: Map<String, DownloadState> = emptyMap(),
     val search: SearchState = SearchState(),
     val undoHistory: UndoHistoryState = UndoHistoryState(),
-    val restoreComplete: Boolean = false
+    val restoreComplete: Boolean = false,
+    val locale: Locale? = null
```

Comment:
Have you considered using some other type here e.g. a String representation of Locale? Ideally, we only have immutable types in the state that don't allow for any side-effects. Locale isn't ideal, but not a big problem.